src/State/Util/ParameterParserTrait.php

```diff
@@ -23,31 +24,54 @@
 trait ParameterParserTrait
 {
     /**
-     * @param array<string, mixed> $values
+     * @param array<string, mixed> $context
+     *
+     * @return array<string, mixed>
      */
-    private function getParameterFlattenKey(string $key, array $values): string
+    private function getParameterValues(Parameter $parameter, ?Request $request, array $context): array
     {
-        $parsedKey = explode('[:property]', $key);
-
-        if (isset($parsedKey[0]) && isset($values[$parsedKey[0]])) {
-            return $parsedKey[0];
+        if ($request) {
+            return ($parameter instanceof HeaderParameterInterface ? $request->attributes->get('_api_header_parameters') : $request->attributes->get('_api_query_parameters')) ?? [];
         }
 
-        return $key;
+        return $context['args'] ?? [];
     }
 
     /**
-     * @param array<string, mixed> $context
+     * @param array<string, mixed> $values
      *
-     * @return array<string, mixed>
+     * @return array<mixed, mixed>|ParameterNotFound|array
      */
-    private function extractParameterValues(Parameter $parameter, ?Request $request, array $context): array
+    private function extractParameterValues(Parameter $parameter, array $values): string|ParameterNotFound|array
     {
-        if ($request) {
-            return ($parameter instanceof HeaderParameterInterface ? $request->attributes->get('_api_header_parameters') : $request->attributes->get('_api_query_parameters')) ?? [];
+        $accessors = null;
+        $key = $parameter->getKey();
+        $parsedKey = explode('[:property]', $key);
+        if (isset($parsedKey[0]) && isset($values[$parsedKey[0]])) {
+            $key = $parsedKey[0];
+        } elseif (str_contains($key, '[')) {
+            preg_match_all('/[\w\-_]+/', $key, $matches);
```

Comment:
```suggestion
            preg_match_all('/[^\[\]]+/', $key, $matches);
```

What do you think about using negated character class, as this would be more bullet proof on complex string (special characters, emojis, ...)
Example which would probably be failing currently: `test😀[a]`, `test.test[a]`, `test@test[a]` ...